src/tike/ptycho/fresnel.py

```diff
@@ -0,0 +1,312 @@
+# This is the python version code for initialize probes
+
+import numpy as np
+
+
+def single_probe(probe_shape, lambda0, dx_dec, dis_defocus, dis_StoD,
+                 zone_plate_params):
+    """Estimate the probe using Fresnel propagation model of focusing optics.
+
+    Example
+    -------
+
+    .. code-block:: python
+
+        single_probe(
+            probe_shape=64
+            lambda0=1.24e-9 / 10
+            dx_dec=75e-6
+            dis_defocus=800e-6
+            dis_StoD=2
+            zone_plate_params=dict(
+                radius=150e-6 / 2,
+                outmost=50e-9,
+                beamstop=60e-6,
+            ),
+        )
+
+    Parameters
+    ----------
+    probe_shape : int
+        The pixel width and height of the (square) probe illumination.
+    lambda0 : float [m]
+        The central wavelength of the illumination.
+    dx_dec : float [m]
+        The pixel size of the detector.
+    dis_defocus : float [m]
+        The defocus distance; the distance from the sample to the focal plane.
+        May be negative.
+    dis_StoD : float [m]
+        The sample to detector distance.
+    zone_plate_params: str or dict
+        One of 'velo', '2idd', 'lamni' or a dictionary with the following keys:
+        radius: float [m] zone plate radius, outmost : float [m] outmost zone
+        width, and beamstop : float [m] diameter of central beamstop.
+
+    Returns
+    -------
+    probe : (1, 1, SHARED, WIDE, HIGH) complex64
+        An estimate of the probe.
+    """
+    # pixel size on sample plane
+    dx = lambda0 * dis_StoD / probe_shape / dx_dec
```

Comment:
Would it make more sense to accept `dx` as an argument directly instead of using `dis_StoD` to compute dx?